code/gui/griddisplay.cpp

```diff
@@ -1,62 +1,97 @@
 }
 
 void GridDisplay::buttonClicked(int x, int y) {
-    if (squareSelected[x][y] > notSelectedWeight){
+    if (startPosSelected) {
+        squareSelected[x][y] = startWeight;
+        m_button[x][y]->setStyleSheet(startSelectedStyle);
+        m_start_position->x = x;
+        m_start_position->y = y;
+        startPosSelected = false;
+#ifndef NDEBUG
+        qDebug() << "Robot Start Position (" << x << "," << y << ") = " << squareSelected[x][y];
+#endif
+    } else if (endPosSelected) {
+        if (m_end_position->x != notSelectedWeight && m_end_position->y != notSelectedWeight) { //If another end position was previously selected
+            squareSelected[m_end_position->x][m_end_position->y] = -1;
+            m_button[m_end_position->x][m_end_position->y]->setStyleSheet(buttonStyle);
+        }
+        squareSelected[x][y] = endWeight;
+        m_button[x][y]->setStyleSheet(endSelectedStyle);
+        m_end_position->x = x;
+        m_end_position->y = y;
+        endPosSelected = false;
+#ifndef NDEBUG
+        qDebug() << "Robot End Position (" << x << "," << y << ") = " << squareSelected[x][y];
+#endif
+    } else if (squareSelected[x][y] > notSelectedWeight
+           || squareSelected[x][y] == startWeight
+           || squareSelected[x][y] == endWeight) {
         squareSelected[x][y] = notSelectedWeight;
         m_button[x][y]->setStyleSheet(buttonStyle);
     } else {
-        squareSelected[x][y] = defaultWeight;
-        m_button[x][y]->setStyleSheet(buttonSelectedStyle);
+        squareSelected[x][y] = m_camera_display->getWeighting();
+       // m_button[x][y]->setStyleSheet(buttonSelectedStyle);
+        //sets button to different shades of green based on weighting assigned
+        m_button[x][y]->setStyleSheet(buttonSelectedStyle.arg(255 - 10*m_camera_display->getWeighting()));
     }
 #ifndef NDEBUG
     qDebug() << "Button (" << x << "," << y << ") = " << squareSelected[x][y];
 #endif
 }
 
 void GridDisplay::drawButtons() {
-//    m_signalmapper = new QSignalMapper(this);
-//    connect(m_signalmapper, SIGNAL(mapped(int)), this, SIGNAL(buttonClicked(int)));
     for (int y = 0; y < rowCount; y++) {
         for (int x = 0; x < columnCount; x++) {
             QString text = QString::number(x);
             m_button[x][y] = new QPushButton();
             m_button[x][y]->setGeometry(QRect(x * gridSize, y * gridSize, gridSize, gridSize));
             m_button[x][y]->setStyleSheet(buttonStyle);
             squareSelected[x][y] = notSelectedWeight;
-            //multiple signals solution 1: signal mapper
-//        m_signalmapper->setMapping(m_button[i], i);
-//        connect(m_button[i], SIGNAL(clicked()), m_signalmapper, SLOT(map()));
-            //multiple signals solution 2: sender() function
-            //connect(m_button[i], SIGNAL(clicked()), this, SLOT(buttonClicked()));
-
-            //multiple signals solution 3: lambda functions
+            //lambda function to receive signals from multiple buttons
```

Comment:
Comments that aren't commenting out code should have a space between the `//` and the text that follows